default/AI/ResourcesAI.py

```diff
@@ -117,6 +124,87 @@ def print_resources_priority():
     warnings.clear()
 
 
+def weighted_sum_output(outputs):
+    """Return a weighted sum of planetary output
+    :param outputs: (industry, research)
+    :return: weighted sum of industry and research
+    """
+    return outputs[0] + RESEARCH_WEIGHTING * outputs[1]
+
+
+def assess_protection_focus(pid):
+    """Return True if planet should use Protection Focus"""
+    this_planet = planetMap[pid]
+    sys_status = foAI.foAIstate.systemStatus.get(this_planet.systemID, {})
+    threat_from_supply = (0.25 * foAI.foAIstate.empire_standard_enemy_rating *
+                          min(2, len(sys_status.get('enemies_nearly_supplied', []))))
+    print "Planet %s has regional+supply threat of %.1f" % ('P_%d<%s>'%(pid, this_planet.name), threat_from_supply)
+    regional_threat = sys_status.get('regional_threat', 0) + threat_from_supply
+    if not regional_threat:  # no need for protection
+        if currentFocus[pid] == PFocus:
+            print "Advising dropping Protection Focus at %s due to no regional threat" % this_planet
+        return False
+    cur_prod_val = weighted_sum_output([currentOutput[pid][IFocus], currentOutput[pid][RFocus]])
+    target_prod_val = max(map(weighted_sum_output, [newTargets[pid][IFocus], newTargets[pid][RFocus]]))
+    prot_prod_val = weighted_sum_output(newTargets[pid][PFocus])
+    local_production_diff = 0.8 * cur_prod_val + 0.2 * target_prod_val - prot_prod_val
+    fleet_threat = sys_status.get('fleetThreat', 0)
+    # TODO: relax the below rejection once the overall determination of PFocus is better tuned
+    if not fleet_threat and local_production_diff > 8:
+        if currentFocus[pid] == PFocus:
+            print "Advising dropping Protection Focus at %s due to excessive productivity loss" % this_planet
+        return False
+    local_p_defenses = sys_status.get('mydefenses', {}).get('overall',0)
+    # TODO have adjusted_p_defenses take other in-system planets into account
+    adjusted_p_defenses = local_p_defenses * (1.0 if currentFocus[pid] != PFocus else 0.5)
+    local_fleet_rating = sys_status.get('myFleetRating', 0)
+    combined_local_defenses = sys_status.get('all_local_defenses', 0)
+    my_neighbor_rating = sys_status.get('my_neighbor_rating', 0)
+    neighbor_threat = sys_status.get('neighborThreat', 0)
+    safety_factor = 1.2 if currentFocus[pid] == PFocus else 0.5
+    cur_shield = this_planet.currentMeterValue(fo.meterType.shield)
+    max_shield = this_planet.currentMeterValue(fo.meterType.maxShield)
```

Comment:
IMHO it is ok to make each small fix to separate commit and push it.  First review I usually start form whole diff, but for fixes I see commits. Since you published you code, please, push often, because somebody can review outdated code. 

For good history we can resubmit final version as another pr as we already do with design.

PS. This group of comment will not be hidden, because you remove other line :)